### DISCUSSION TITLE
diesel print-schema > src/schema.rs doesn't contain diesel::joinable! macro for foreign keys (Mysql)

### DISCUSSION THREAD

**Discussion by minttr** [diesel-rs/diesel#3493] [Q&A]
I have created two tables in my `Mysql` db using following instructions:
```
CREATE TABLE user (
    UserID int NOT NULL,
    UserName varchar(100),
    CONSTRAINT PK_USER PRIMARY KEY (UserID)
);
```
and
```
CREATE TABLE app (
    AppID int NOT NULL,
    AppName varchar(100),
    AppTitle varchar(100),
    AppOwner int,
    CONSTRAINT PK_APP PRIMARY KEY (AppID),
    CONSTRAINT FK_USER FOREIGN KEY (AppOwner) REFERENCES user(UserID)
);
```

I want the `app` table to contain foreign key `AppOwner` which references `userID` in table `user`.

Then when I generate `schema.rs` file using `diesel print-schema > src/schema.rs`, the out out is:
```
// @generated automatically by Diesel CLI.

diesel::table! {
    app (AppID) {
        AppID -> Integer,
        AppName -> Nullable<Varchar>,
        AppTitle -> Nullable<Varchar>,
        AppOwner -> Nullable<Integer>,
    }
}

diesel::table! {
    user (UserID) {
        UserID -> Integer,
        UserName -> Nullable<Varchar>,
    }
}

diesel::allow_tables_to_appear_in_same_query!(
    app,
    user,
);
```

I deally it should have also inserted `diesel::joinable!(app -> user(AppOwner));` to enable joining. Am I missing something here?

**Comment by weiznich** [comment]
It's hard to tell what's actually wrong here. I've tried the provided schema on my local installation and got the following `schema.rs` file (which includes the `joinable!()` call):

```rust
// @generated automatically by Diesel CLI.

diesel::table! {
    app (AppID) {
        AppID -> Integer,
        AppName -> Nullable<Varchar>,
        AppTitle -> Nullable<Varchar>,
        AppOwner -> Nullable<Integer>,
    }
}

diesel::table! {
    user (UserID) {
        UserID -> Integer,
        UserName -> Nullable<Varchar>,
    }
}

diesel::joinable!(app -> user (AppOwner));

diesel::allow_tables_to_appear_in_same_query!(
    app,
    user,
);
```

For reference thats with:
* `diesel-cli`: 2.0.1
* `MariaDB`: 10.5.18

It's likely that this behavior depends on your database version as `diesel` just queries the database for the relevant information. So the relevant location might have changed for your database version.
